internal/tests/testrepository/header_testrepository.go

```diff
@@ -193,6 +195,70 @@ func (r *HeaderTestRepository) GetChainBetweenTwoHashes(low string, high string)
 	return headers, nil
 }
 
+// GetMerkleRoots returns ExclusiveStartKey pagination of batchSize size with merkle roots from lastEvaluatedKey which
+// is the last height of the block that a client has processed
+func (r *HeaderTestRepository) GetMerkleRoots(batchSize int, lastEvaluatedKey string) (*domains.MerkleRootsESKPagedResponse, error) {
+	merkleroots := make([]*domains.BlockHeader, 0)
+	// order headers by height ASC
+	sort.Slice(*r.db, func(i, j int) bool {
+		return (*r.db)[i].Height < (*r.db)[j].Height
+	})
+
+	// initial value indicates start of the chain
+	lastEvaluatedMerklerootHeight := int32(-1)
+	idx := slices.IndexFunc(*r.db, func(c domains.BlockHeader) bool { return c.MerkleRoot.String() == lastEvaluatedKey })
+
+	if idx == -1 && lastEvaluatedKey != "" {
+		return nil, errors.New(domains.MerklerootNotFoundError)
+	}
+
+	if idx != -1 {
+		lastEvaluatedMerkleroot := (*r.db)[idx]
+		lastEvaluatedMerklerootHeight = lastEvaluatedMerkleroot.Height
+	}
+
+	// include only those headers that height is greater than merkleroot of lastEvaluatedKey
+	for _, header := range *r.db {
+		if header.Height > lastEvaluatedMerklerootHeight {
+			hd := header
+			merkleroots = append(merkleroots, &hd)
+		}
+	}
```

Comment:
You could take advantage of golang `slices` here 🙈 
Please validate my idea, but here it is:
- the `*r.db` is sorted by `Height`
- so, find the startingIndex (using your own function or `slices.IndexFunc`)
- make a slice `merkleroots := (*r.db)[startingIndex:]`
- just like that - we would avoid appending to the new "slice"

https://gobyexample.com/slices